### PR TITLE
Add edge provider

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
Adding a new cloud provider(edge) which is responsible for joining edge devices to KKP user clusters. The edge cloud provider is a bit similar to the Kubeadm provide however it has more capabilities such as configuring the machines that will be joining the cluster by using OSM. 
 
**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind feature
/kind api-change

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Adding the edge cloud provider
```

**Documentation**:

```documentation
None
```

### CODE REVIEW

**Review by ahmedwaleedmalik — COMMENTED** [review]
Overall it looks good, just a few comments:
1. There are failing tests
2. Are we sure that we want to ship this in KKP 2.25? Since at the moment it's just bare bone skeleton for the provider and actual implementation, coming from OSM and MC(probably) is sitll not there.

File: pkg/resources/apiserver/deployment.go [hunk 1]
```diff
@@ -339,7 +339,7 @@ func getApiserverFlags(data *resources.TemplateData, etcdEndpoints []string, ena
 
 	// the "bring-your-own" provider does not support automatic TLS rotation in kubelets yet,
 	// and because of that certs might expire and kube-apiserver cannot validate the connection anymore.
-	if cluster.Spec.Cloud.BringYourOwn == nil {
+	if cluster.Spec.Cloud.BringYourOwn == nil || cluster.Spec.Cloud.Edge == nil {
```
Comment (embik) [inline pkg/resources/apiserver/deployment.go]: I can't spot what it is to be honest, but there's something wrong here. The fixture changes below for `bringyourown` are not desired, `--kubelet-certificate-authority` must not be set (#9674).

**Comment by moadqassem** [comment]
> Overall it looks good, just a few comments:
> 
> 1. There are failing tests
> 
> 2. Are we sure that we want to ship this in KKP 2.25? Since at the moment it's just bare bone skeleton for the provider and actual implementation, coming from OSM and MC(probably) is sitll not there.

I agree that this looks quite vague 😅. The actually implementation is gonna be actually in the API and the dashboard but to do that I need this first. Regarding the falling test will take a look at it later today.

File: pkg/resources/apiserver/deployment.go (hunk 1 above)
Comment (moadqassem) [reply pkg/resources/apiserver/deployment.go:342]: Ah I see, you mean that those files should not have been generated! I guess the condition is wrong. It should be && not ||

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Pull request merged**